src/LogFileReader.cpp

```diff
@@ -74,12 +79,32 @@ bool LogFileReader::readAll(const QString& fileName)
 // File input is a csv file in the format of hh:mm:ss:zzz, voltage, current
 bool LogFileReader::parseLine(const QString& line, BatteryData& batteryData) const
 {
-    // TODO implement this first
+    QStringList lines = line.split(COMMA);
+    QTime Time = QTime::fromString(lines[0], TIMEFORMAT);
+    bool ok;
+    if (Time.isValid())
+    {
+        batteryData.time = Time;
+    }
+    else
+    {
+        return false;
+    }
 
-    // This is here to the compiler happy. Otherwise the compile
-    // will have an error warning about an unused variable. Remove this
-    // when you use it.
-    Q_UNUSED(line);
-    Q_UNUSED(batteryData);
+    if (lines.count() !=    3)
+    {
+        return false;
+    }
+
+    for (int c = 1; c<=2; c++)
```

Comment:
It's much more standard to use `i` in for loops. The `c` has potential to confuse people to look for context where there isnt' any.